packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -4823,6 +4823,53 @@ class RenderLeaderLayer extends RenderProxyBox {
   }
 }
 
+/// A class that can supply the transform used in a
+/// [RenderFollowerLayer.transformDelegate] or
+/// [CompositedTransformFollower.transformDelegate].
+///
+/// Create a subclass of this to override the default computation of the offset
+/// for a [FollowerLayer] (or [RenderFollowerLayer]) in [computeOffset].
+///
+/// {@template flutter.rendering.proxy_box.follower_layer_transform_delegate_note}
+/// This is used to provide a custom transform when calculating the transform
+/// for a [CompositedTransformFollower], which can perform transforms during the
+/// paint phase of the rendering pipeline to render objects that come later in
+/// the paint order, instead of having to wait an additional frame to update the
+/// follower's transform, as would normally have to happen if synchronizing two
+/// widgets in different parts of the tree.
+///
+/// This is often used to synchronize elements in an [Overlay] with elements in
+/// the main widget tree.
+/// {@endtemplate}
+@immutable
+class FollowerLayerTransformDelegate {
+  /// Creates a const [FollowerLayerTransformDelegate].
+  ///
+  /// This is declared as const so that subclasses may be const.
+  const FollowerLayerTransformDelegate();
+
+  /// Computes the offset of a [FollowerLayer] based on the input alignments and
+  /// sizes.
+  ///
+  /// The default implementation honors the given alignments and offsets,
+  /// aligning the `followerAnchor` point with the `leaderAnchor` point, offset
+  /// by `offset`.
+  ///
+  /// The `leaderSize` argument will only be null when `leaderAnchor` is
+  /// [Alignment.topLeft], or when no leader is linked.
+  Offset computeOffset({
```

Comment:
Wondering: Can this just be specified on RenderFollowerLayer/CompositedTransformFollower as a callback instead of requiring a subclass?

Reply:
Yes, it could be. My reasoning on this was that the word "transform" in the name meant that we could someday add scaling or rotation to the transformations in addition to offsets, and then it seemed like you'd want to have one subclass instead of multiple callbacks.

Also, for my use case, it was helpful for it to be a subclass that could hold state used to compute the offset, and if it was just a callback, I'd have to make a class anyhow to hold the state and keep the instance somewhere and pass a call to that object's function for computing it, so this was easier for my use case, which makes a weak case (weak because of a sample size of 1) that this would be useful for others too.